packages/core/src/hydration/api.ts

```diff
@@ -88,8 +95,23 @@ function printHydrationStats(injector: Injector) {
  * Returns a Promise that is resolved when an application becomes stable.
  */
 function whenStable(
-    appRef: ApplicationRef, pendingTasks: InitialRenderPendingTasks): Promise<unknown> {
+    appRef: ApplicationRef, pendingTasks: InitialRenderPendingTasks,
+    injector: Injector): Promise<unknown> {
   const isStablePromise = appRef.isStable.pipe(first((isStable: boolean) => isStable)).toPromise();
+  if (typeof ngDevMode !== 'undefined' && ngDevMode) {
+    const timeoutTime = APPLICATION_IS_STABLE_TIMEOUT;  // could be provided via token
```

Comment:
I'd propose dropping this comment for now:
```suggestion
    const timeoutTime = APPLICATION_IS_STABLE_TIMEOUT;
```